tests/validation/tests/v3_api/resource/terraform/charts/do/main.tf

```diff
@@ -0,0 +1,176 @@
+#Cluster creation with v1 charts installed
+terraform {
+  required_version = ">= 0.14"
+  required_providers {
+    rancher2 = {
+      source = "rancher/rancher2"
+    }
+
+  }
+}
+
+provider "rancher2" {
+  api_url   = var.rancher_api_url
+  token_key = var.rancher_token_key
+}
+
+resource "rancher2_cluster" "custom" {
+  name = "do-tf-charts"
+  description = "Rancher2 TF Custom Cluster"
+  rke_config {
+    network {
+      plugin = "canal"
+    }
+  }  
+  enable_cluster_monitoring = true
+  cluster_monitoring_input {
+    answers = {
+      "exporter-kubelets.https" = true
+      "exporter-node.enabled" = true
+      "exporter-node.ports.metrics.port" = 9796
+      "exporter-node.resources.limits.cpu" = "200m"
+      "exporter-node.resources.limits.memory" = "200Mi"
+      "grafana.persistence.enabled" = false
+      "grafana.persistence.size" = "10Gi"
+      "grafana.persistence.storageClass" = "default"
+      "operator.resources.limits.memory" = "500Mi"
+      "prometheus.persistence.enabled" = "false"
+      "prometheus.persistence.size" = "50Gi"
+      "prometheus.persistence.storageClass" = "default"
+      "prometheus.persistent.useReleaseName" = "true"
+      "prometheus.resources.core.limits.cpu" = "1000m",
+      "prometheus.resources.core.limits.memory" = "1500Mi"
+      "prometheus.resources.core.requests.cpu" = "750m"
+      "prometheus.resources.core.requests.memory" = "750Mi"
+      "prometheus.retention" = "12h"
+    }
+  }
+}
+
+resource "rancher2_node_template" "node-template" {
+  name = "tf-do-node-template"
+  description = "TF DO node template"
+  engine_install_url = "https://releases.rancher.com/install-docker/20.10.sh"
+  
+  digitalocean_config {
+    access_token = var.do_access_token
+    region = var.region
+    size = var.size
+    image = var.image
+  }
+}
+
+resource "rancher2_node_pool" "node-pool" {
+  cluster_id =  rancher2_cluster.custom.id
+  name = "tf-node-pool"
+  hostname_prefix = var.hostname_prefix
+  node_template_id = rancher2_node_template.node-template.id
+  quantity = 3
+  control_plane = true
+  etcd = true
+  worker = true
+}
+
+resource "rancher2_cluster_sync" "custom-sync" {
+  cluster_id = rancher2_cluster.custom.id
+}
+
+resource "rancher2_namespace" "istio-namespace" {
+  name = "istio-system"
+  project_id = rancher2_cluster_sync.custom-sync.system_project_id
+  description = "istio namespace"
+}
+
+resource "rancher2_app" "istio" {
+  catalog_name = "system-library"
+  name = "cluster-istio"
+  description = "Terraform app acceptance test"
+  project_id = rancher2_cluster_sync.custom-sync.system_project_id
+  template_name = "rancher-istio"
+  template_version = "1.5.901"
```

Comment:
We should probably parameterize the template_version for each app and set a sensible default. Maybe default to null if `rancher2_app` defaults to using the latest available?